File: frontend/src/app/modules/models/customerOrErrorsModel.ts
import {customerModel} from "./customerModel";

export class customerOrErrorsModel {
  customerModel: customerModel = new customerModel();
  errors: Map<string, string> = new Map<string, string>();


  public constructor() {
    this.errors = new Map<string, string>();
    this.customerModel = new customerModel();
  }

  static cloneBase(customerOrErrors: customerOrErrorsModel): customerOrErrorsModel {
    const clonedCustomerOrErrors: customerOrErrorsModel = new customerOrErrorsModel();
    clonedCustomerOrErrors.customerModel = customerOrErrors.customerModel;
    clonedCustomerOrErrors.errors = customerOrErrors.errors;
    return clonedCustomerOrErrors;
  }
}

File: fapi/src/main/java/com/artsykov/fapi/models/CustomerOrErrorsModel.java
package com.artsykov.fapi.models;

import lombok.Data;
import org.springframework.stereotype.Component;

import java.util.Map;

@Data
public class CustomerOrErrorsModel {
    private CustomerModel customerModel;
    private Map<String, String> errors;

    public CustomerOrErrorsModel() {}

    public CustomerOrErrorsModel(Map<String, String> errors) {
        this.errors = errors;
    }

    public CustomerOrErrorsModel(CustomerModel customerModel) {
        this.customerModel = customerModel;
    }
}
